bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/StreamingBigtableResultScanner.java

```diff
@@ -124,10 +125,12 @@ public void addResult(ReadRowsResponse response) {
   }
 
   public void setError(Throwable error) {
+    reservedChannel.returnToPool();
```

Comment:
The channel will get closed when the BigtableSession closes.  We have hooks for that.  We don't actually create new Channel in the pool.  It's more likely that if we have lots of scans that somehow lose channels, that we'll go down from 4 channels to 1, and we'll have a potential performance hit.